plugins/GSdx/GSTextureCacheSW.cpp

```diff
@@ -133,30 +121,26 @@ void GSTextureCacheSW::RemoveAll()
 
 void GSTextureCacheSW::IncAge()
 {
-	for(hash_set<Texture*>::iterator i = m_textures.begin(); i != m_textures.end(); )
+	for(auto i = m_textures.begin(); i != m_textures.end(); )
 	{
-		hash_set<Texture*>::iterator j = i++;
-
-		Texture* t = *j;
+		Texture* t = *i;
 
 		if(++t->m_age > 10)
 		{
-			m_textures.erase(j);
+			i = m_textures.erase(i);
```

Comment:
Just for reference, previous pattern is:

```C++
for (auto i = list.begin(); i != list.end(); ) {
    auto j = i++;
    if (condition) {
        list.erase(j);
    }
}
```
Well this solution has one more variable but no else branch and discards `erase` return value. Still convinced that it's better? I just thought that the return value of `erase` should be spoiled, as it let room for optimization, but I let you decide.